server/src/main/java/info/doseamigos/meds/MedSpeechlet.java

```diff
@@ -0,0 +1,107 @@
+package info.doseamigos.meds;
+
+import com.amazon.speech.slu.Intent;
+import com.amazon.speech.speechlet.*;
+import com.amazon.speech.ui.PlainTextOutputSpeech;
+import com.amazon.speech.ui.Reprompt;
+import com.amazon.speech.ui.SimpleCard;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.inject.Guice;
+import com.google.inject.Injector;
+import org.apache.http.HttpResponse;
+import org.apache.http.client.HttpClient;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.impl.client.HttpClients;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Speechlet for adding medication.
+ */
+public class MedSpeechlet implements Speechlet {
+
+    private static final Logger log = LoggerFactory.getLogger(MedSpeechlet.class);
+    private MedService medService;
+
+    @Override
+    public void onSessionStarted(SessionStartedRequest request, Session session) throws SpeechletException {
+        Injector injector = Guice.createInjector(
+            new MedGuiceModule()
+        );
+        medService = injector.getInstance(MedService.class);
+    }
+
+    @Override
+    public SpeechletResponse onLaunch(LaunchRequest request, Session session) throws SpeechletException {
+        log.info(session.getUser().getUserId());
+        if (session.getUser().getAccessToken() == null) {
+            log.info("The access token is null.");
+            throw new NullPointerException();
+        }
+        String accessToken = session.getUser().getAccessToken();
+        Map<String, String> userInfo = null;
+        try {
+            log.info("Attempting to get user from access token");
+            userInfo = getUserInfo(accessToken);
+
+        } catch (IOException e) {
+            log.error("error occurred grabbing user info from google", e);
+        }
+        log.info("Received user from google.");
+        String name = userInfo == null ? "John Doe" : userInfo.get("name");
```

Comment:
This was how I tested if this worked or not.  Something I should create a ticket for is error handling.  Something the echo likes to do is de-authenticate and make me disable/reenable the skill to re-authenticate with Google.  It happens about every couple of hours or so.  We should handle that cleanly.